### ISSUE TITLE
storageclass cinder is ignoring availability zone

### ISSUE THREAD

**Issue by zetaab** [kubernetes/kubernetes#44032]
**Is this a BUG REPORT or FEATURE REQUEST?** BUG report

**Kubernetes version** (use `kubectl version`):
kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.0", GitCommit:"fff5156092b56e6bd60fff75aad4dc9de6b6ef37", GitTreeState:"clean", BuildDate:"2017-03-28T16:36:33Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-1+c0b74ebf3ce26e-dirty", GitCommit:"c0b74ebf3ce26e46b5397ffb5ce71cd02f951130", GitTreeState:"dirty", BuildDate:"2017-03-30T08:19:21Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Cloud provider or hardware configuration**: openstack
- **OS** (e.g. from /etc/os-release): centos 7
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

I created new cinder storageclass, however it does ignore availability zone and it will provision volumes zone called nova (default)

yaml used to create cinder:

```
kind: StorageClass
apiVersion: storage.k8s.io/v1beta1
metadata:
  name: standard
provisioner: kubernetes.io/cinder
parameters:
  availability: zone-1
```

```
% openstack volume list
+--------------------------------------+-------------------------------------------------------------+-----------+------+-------------+
| ID                                   | Display Name                                                | Status    | Size | Attached to |
+--------------------------------------+-------------------------------------------------------------+-----------+------+-------------+
| d71f8af4-2e84-4908-a3e8-b5f64d0ed3aa | kubernetes-dynamic-pvc-810f0f8b-191b-11e7-9a54-fa163e26fe46 | error     |   12 |             |
+--------------------------------------+-------------------------------------------------------------+-----------+------+-------------+

% openstack volume show d71f8af4-2e84-4908-a3e8-b5f64d0ed3aa
+------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Field                        | Value                                                                                                                                                                                   |
+------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| attachments                  | []                                                                                                                                                                                      |
| availability_zone            | nova                                                                                                                                                                                    |
| bootable                     | false                                                                                                                                                                                   |
| consistencygroup_id          | None                                                                                                                                                                                    |
| created_at                   | 2017-04-04T09:56:43.000000                                                                                                                                                              |
| description                  | None                                                                                                                                                                                    |
| encrypted                    | False                                                                                                                                                                                   |
| id                           | d71f8af4-2e84-4908-a3e8-b5f64d0ed3aa                                                                                                                                                    |
| multiattach                  | False                                                                                                                                                                                   |
| name                         | kubernetes-dynamic-pvc-810f0f8b-191b-11e7-9a54-fa163e26fe46                                                                                                                             |
| os-vol-tenant-attr:tenant_id | b23efb65b1d44b5abd561511f40c565d                                                                                                                                                        |
| properties                   | kubernetes.io/created-for/pv/name='pvc-810f0f8b-191b-11e7-9a54-fa163e26fe46', kubernetes.io/created-for/pvc/name='storage-es-data-0', kubernetes.io/created-for/pvc/namespace='default' |
| replication_status           | disabled                                                                                                                                                                                |
| size                         | 12                                                                                                                                                                                      |
| snapshot_id                  | None                                                                                                                                                                                    |
| source_volid                 | None                                                                                                                                                                                    |
| status                       | error                                                                                                                                                                                   |
| type                         | None                                                                                                                                                                                    |
| updated_at                   | 2017-04-04T09:56:43.000000                                                                                                                                                              |
| user_id                      | 56ebdadedcbd34ccac211665a0bf5d78060040f1ff05fb5e896d77e7f882650d                                                                                                                        |
+------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

```

I added manual print here https://github.com/kubernetes/kubernetes/blob/v1.6.0/pkg/cloudprovider/providers/openstack/openstack_volumes.go#L145

It prints out:
```
{"log":"I0404 11:08:15.852302       1 openstack_volumes.go:145] Createvolume availability: zone-1\n","stream":"stderr","time":"2017-04-04T11:08:15.852382116Z"
```

I tried to remove that newline from that string, but still it is creating that to nova availability zone. For me this looks like gophercloud volumes.CreateOpts{ is not working as should in createvolume function https://github.com/kubernetes/kubernetes/blob/v1.6.0/pkg/cloudprovider/providers/openstack/openstack_volumes.go#L144-L171 
**What you expected to happen**:
I except that if I define availability, like docs says it should also use that parameter when provisioning volume.

**How to reproduce it** (as minimally and precisely as possible):

```
kind: StorageClass
apiVersion: storage.k8s.io/v1beta1
metadata:
  name: standard
provisioner: kubernetes.io/cinder
parameters:
  availability: zone-1
```

```
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  name: es-data
  labels:
    component: elasticsearch
    role: data
spec:
  serviceName: elasticsearch-data
  replicas: 3
  template:
    metadata:
      labels:
        component: elasticsearch
        role: data
      annotations:
        pod.beta.kubernetes.io/init-containers: '[
          {
          "name": "sysctl",
            "image": "busybox",
            "imagePullPolicy": "IfNotPresent",
            "command": ["sysctl", "-w", "vm.max_map_count=262144"],
            "securityContext": {
              "privileged": true
            }
          }
        ]'
    spec:
      containers:
      - name: es-data
        securityContext:
          privileged: true
          capabilities:
            add:
              - IPC_LOCK
        image: quay.io/pires/docker-elasticsearch-kubernetes:5.2.2_2
        imagePullPolicy: Always
        env:
        - name: NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: "CLUSTER_NAME"
          value: "myesdb"
        - name: NODE_MASTER
          value: "false"
        - name: NODE_INGEST
          value: "false"
        - name: HTTP_ENABLE
          value: "false"
        - name: "ES_JAVA_OPTS"
          value: "-Xms256m -Xmx256m"
        ports:
        - containerPort: 9300
          name: transport
          protocol: TCP
        volumeMounts:
        - name: storage
          mountPath: /data
      nodeSelector:
        failure-domain.beta.kubernetes.io/zone: zone-1
  volumeClaimTemplates:
  - metadata:
      name: storage
      annotations:
        volume.beta.kubernetes.io/storage-class: standard
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 12Gi
```

**Anything else we need to know**:

**Comment by zetaab** [comment]
possible configuration problem in our openstack

**Comment by zetaab** [comment]
gophercloud is posting following to openstack API:
```
{"volume":{"availability":"zone-1","display_name":"kubernetes-dynamic-pvc-db00e45a-1933-11e7-ad07-fa163e07a592","metadata":{"kubernetes.io/created-for/pv/name":"pvc-db00e45a-1933-11e7-ad07-fa163e07a592","kubernetes.io/created-for/pvc/name":"storage-es-data-0","kubernetes.io/created-for/pvc/namespace":"default"},"size":12}}
```

When creating volume manually using `openstack volume create --availability-zone zone-1 --size 12 foobar --debug`

the data is
```
{"volume": {"status": "creating", "user_id": null, "name": "foobar", "imageRef": null, "availability_zone": "zone-1", "description": null, "multiattach": false, "attach_status": "detached", "volume_type": null, "metadata": {}, "consistencygroup_id": null, "source_volid": null, "snapshot_id": null, "project_id": null, "source_replica": null, "size": 12}}
```

So we can clearly see that gophercloud is using wrong variable name here, it should be `availability_zone` instead of `availability`

**Comment by zhouhaibing089** [comment]
cc @anguslees

**Comment by zetaab** [comment]
This is actually fixed already in 1.6.1 https://github.com/kubernetes/kubernetes/blob/v1.6.1/vendor/github.com/rackspace/gophercloud/openstack/blockstorage/v1/volumes/requests.go

**Comment by xsgordon** [comment]
/cc @k8s-sig-openstack-bugs @k8s-mirror-storage-bugs 

@idvoretskyi can you please label with sig/openstack and sig/storage please?
